Core/GDCore/IDE/Events/EventsVariableReplacer.cpp

```diff
@@ -364,6 +365,16 @@ bool EventsVariableReplacer::DoVisitInstruction(gd::Instruction& instruction,
                 parameterIndex, ExpressionParser2NodePrinter::PrintNode(*node));
           }
         }
+
+        if (gd::ParameterMetadata::IsExpression("variable", type)) {
+          const auto &variableName = instruction.GetParameter(parameterIndex).GetPlainString();
+          auto itr = variableNewTypes.find(variableName);
```

Comment:
A "variable name" is an expression, for example it can be `MyVariable.MyChild`. In this case, it won't work right? What are the implications of this?